excel.go

```diff
@@ -0,0 +1,217 @@
+package main
+
+import (
+	"github.com/xuri/excelize/v2"
+)
+
+const (
+	user = iota
+	portal
+	previous
+	timestamp
+	numCols        int    = 4
+	automatedSheet string = "PasswordManager"
+)
+
+func getMCFinUsers(f *excelize.File, config *Portal) (map[string]string, error) {
+	rows, err := f.GetRows(config.SheetName)
+	if err != nil {
+		config.errorLog.Printf("failed getting rows from %s in %s", config.SheetName, config.Filename)
+		return nil, err
+	}
+
+	users := make(map[string]string)
+
+	headerToXCoord := getHeaderToXCoord(rows[0])
+	usernameXCoord := headerToXCoord[config.UsernameHeader]
+	passwordXCoord := headerToXCoord[config.PasswordHeader]
+
+	for _, row := range rows[rowOffset:] {
+		// check if row is empty
+		if row == nil {
+			continue
+		}
+		users[row[usernameXCoord]] = row[passwordXCoord]
+	}
+
+	return users, nil
+}
+
+func getPasswordManagerUsers(f *excelize.File, config *Portal) (map[string]string, error) {
+	rows, err := f.GetRows(automatedSheet)
+	if err != nil {
+		config.errorLog.Printf("failed getting rows from %s in %s", automatedSheet, config.Filename)
+		return nil, err
+	}
+
+	passwordManagerUsersToPassword := make(map[string]string)
+	for _, row := range rows[rowOffset:] {
+		passwordManagerUsersToPassword[row[user]] = row[portal]
+	}
+
+	return passwordManagerUsersToPassword, nil
+}
+
+// Sync PasswordManager usernames with MACFIN users
+func syncPasswordManagerUsersToMACFINUsers(f *excelize.File, config *Portal) error {
+	macFinUsersToPasswords, err := getMCFinUsers(f, config)
+	if err != nil {
+		return err
+	}
+
+	passwordManagerUsersToPassword, err := getPasswordManagerUsers(f, config)
+	if err != nil {
+		config.errorLog.Printf("failed reading users from %s", automatedSheet)
```

Comment:
nit: this message refers to an implementation detail of `getPasswordManagerUsers`. In any case the error is already being logged so you don't need to log it again.

Reply:
Done. Refactored logging throughout the application.